definitions/checks/katello_agent.rb

```diff
@@ -0,0 +1,39 @@
+require 'yaml'
+
+class Checks::CheckKatelloAgentEnabled < ForemanMaintain::Check
+  metadata do
+    label :check_katello_agent_enabled
+    description 'Check whether the katello-agent feature is enabled before upgrading'
+    tags :pre_upgrade
+
+    confine do
+      feature(:instance).downstream
+    end
+  end
+
+  INSTALLER_ANSWERS_FILE = '/etc/foreman-installer/scenarios.d/satellite-answers.yaml'.freeze
+
+  def run
+    assert(
+      !katello_agent_enabled?,
+      "The katello-agent feature is enabled on this system. As of Satellite 6.15.0, katello-agent"\
+      " is removed and will no longer function. Before proceeding with the upgrade, you should"\
+      " ensure that you have deployed and configured an alternative tool for remote package"\
+      " management and patching for content hosts, such as Remote Execution (REX) with pull-based"\
+      " transport. See the Managing Hosts guide in the Satellite documentation for more info."\
+      " Disable katello-agent with the command"\
+      " `satellite-installer --foreman-proxy-content-enable-katello-agent false`"\
+      " before proceeding with the upgrade."
```

Comment:
This should probably be written in a way that also applies to upstream?

Reply:
I thought upstream users don't use foreman-maintain? Does foreman-maintain have a downstream repo that can be branded?

Reply:
no, this repo is capable of branding itself. like `feature(:installer).installer_command` will give you foreman-installer/satellite-installer as needed.

and while upstream users don't use f-m much for upgrades, the code is capable of doing it (mostly) and we should strive not to break that.

Reply:
oh, and the `satellite-installer` command :yum:

Reply:
It can't do the version translation, but I also think it doesn't have to?
This check makes only sense at two points:
Katello 4.9 → 4.10 upgrade, Satellite 6.14 → 6.15 upgrade, right?
So you could just write "The katello-agent feature is enabled on this system, but it will be removed during the upgrade".